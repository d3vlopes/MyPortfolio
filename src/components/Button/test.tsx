import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'
import { Github } from '@styled-icons/boxicons-logos'

import Button from '.'

const { getByRole, getByTestId } = screen

describe('<Button />', () => {
  it('should render by default', () => {
    renderWithTheme(
      <Button variant="primary" size="medium">
        Button
      </Button>,
    )

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': '#6D59A8',
      'font-size': '2.2rem',
      width: '21.5rem',
      height: '6.2rem',
    })
  })

  it('should render the variant secondary ', () => {
    renderWithTheme(<Button variant="secondary">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': '#83A848',
    })
  })

  it('should render the variant outline ', () => {
    renderWithTheme(<Button variant="outline">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      border: '1px solid #DEE7EA',
      'background-color': 'transparent',
    })
  })

  it('should render expand version', () => {
    renderWithTheme(<Button expand>Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      width: '31.5rem',
      height: '6.2rem',
    })
  })

  it('should render small size', () => {
    renderWithTheme(<Button size="small">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      'font-size': '1.8rem',
      width: '17.5rem',
      height: '5rem',
    })
  })

  it('should render large size', () => {
    renderWithTheme(<Button size="large">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      'font-size': '2.6rem',
      width: '28.8rem',
      height: '7.6rem',
    })
  })

  it('should render an icon version', () => {
    renderWithTheme(
      <Button icon={<Github data-testid="icon" />}>Button</Button>,
    )

    const button = getByRole('button', { name: /button/i })
    const icon = getByTestId('icon')

    expect(button).toBeInTheDocument()
    expect(icon).toBeInTheDocument()
  })

  it('should render Button as a link', () => {
    renderWithTheme(
      <Button as="a" href="/link">
        Button
      </Button>,
    )

    expect(screen.getByRole('link', { name: /button/i })).toHaveAttribute(
      'href',
      '/link',
    )
  })

  it('should up while hover', async () => {
    renderWithTheme(<Button hoverUp>Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyleRule('transform', 'translateY(-0.5rem)', {
      modifier: ':hover',
    })
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(<Button>Button</Button>)

    expect(container.firstChild).toMatchSnapshot()
  })
})
