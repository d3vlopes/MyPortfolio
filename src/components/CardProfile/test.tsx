import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import CardProfile, { CardProfileProps } from '.'

const props: CardProfileProps = {
  image: '/img/me.jpg',
  name: 'Leandro Lopes',
  age: 27,
  city: 'Santa Cruz do Sul - RS',
  liking: 'Programação, design, música, séries e filmes',
  dream:
    'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
}

const { getByRole, getByText } = screen

describe('<CardProfile />', () => {
  it('should render CardProfile', () => {
    const { container } = renderWithTheme(<CardProfile {...props} />)

    const image = getByRole('img', { name: 'Leandro Lopes' })
    const name = getByText('Leandro Lopes')
    const age = getByText('27 anos')
    const city = getByText('Santa Cruz do Sul - RS')
    const liking = getByText('Programação, design, música, séries e filmes')
    const dream = getByText(
      'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
    )

    expect(image).toHaveAttribute('src', '/img/me.jpg')
    expect(name).toBeInTheDocument()
    expect(age).toBeInTheDocument()
    expect(city).toBeInTheDocument()
    expect(liking).toBeInTheDocument()
    expect(dream).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
