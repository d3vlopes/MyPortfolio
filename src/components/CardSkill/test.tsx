import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'
import { Sword } from '@styled-icons/remix-fill/Sword'

import CardSkill from '.'
import mock from './mock'

const { getByTestId, getByRole, getByText } = screen

describe('<CardSkill />', () => {
  it('should render CardSkill', () => {
    renderWithTheme(
      <CardSkill
        icon={<Sword data-testid="icon" />}
        title={mock[0].title}
        description={mock[0].description}
      />,
    )

    const icon = getByTestId('icon')
    const title = getByRole('heading', { name: /desafios/i })
    const description = getByText(
      /Estou sempre em busca de novos desafios que me tirem do meu porto seguro./i,
    )

    expect(icon).toBeInTheDocument()
    expect(title).toBeInTheDocument()
    expect(description).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(
      <CardSkill
        icon={<Sword data-testid="icon" />}
        title={mock[0].title}
        description={mock[0].description}
      />,
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
