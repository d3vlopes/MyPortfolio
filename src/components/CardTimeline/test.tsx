import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import CardTimeline, { CardTimelineProps } from '.'

const props: CardTimelineProps = {
  title: 'CardTimeline',
  content: '<p>Loren ipsum dolor</p>',
}

const { getByRole, getByText } = screen

describe('<CardTimeline />', () => {
  it('should render CardTimeline', () => {
    const { container } = renderWithTheme(<CardTimeline {...props} />)

    const title = getByRole('heading', { name: 'CardTimeline' })
    const content = getByText('Loren ipsum dolor')
    const p = container.querySelector('p')

    expect(title).toBeInTheDocument()
    expect(content).toBeInTheDocument()
    expect(p?.tagName.toLowerCase()).toBe('p')
  })

  it('should match snapshot ', () => {
    const { container } = renderWithTheme(<CardTimeline {...props} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
