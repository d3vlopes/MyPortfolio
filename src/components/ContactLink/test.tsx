import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'
import { Props } from 'react-inlinesvg'

import ContactLink, { ContactLinkProps } from '.'

const { getByLabelText } = screen

const mock: ContactLinkProps = {
  icon: '/img/icon.svg',
  name: 'Github',
  link: 'www.github.com/user/name',
}

jest.mock('react-inlinesvg', () => {
  function ReactInlineSVG({ name }: Props) {
    return <svg aria-label={name} />
  }

  return ReactInlineSVG
})

describe('<ContactLinks />', () => {
  it('should render ContactLinks', () => {
    const { container } = renderWithTheme(<ContactLink {...mock} />)

    const icon = getByLabelText(mock.name)
    const a = container.querySelector('a')

    expect(icon).toBeInTheDocument()
    expect(a).toHaveAttribute('href', mock.link)
  })
})
