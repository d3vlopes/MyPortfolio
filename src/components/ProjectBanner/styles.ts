import styled, { css } from 'styled-components'

import { Wrapper as ButtonStyles } from 'components/Button/styles'
import { Wrapper as HeadingStyles } from 'components/Heading/styles'

type CoverProps = {
  src: string
}

export const Cover = styled.div<CoverProps>`
  ${({ src, theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    height: 60rem;
    background-image: linear-gradient(
        177.39deg,
        rgba(0, 0, 0, 0.2) 45%,
        rgba(0, 0, 0, 0) 100%
      ),
      url(${src});
    background-size: cover;
    background-position: top center;
    background-repeat: no-repeat;

    ${HeadingStyles} {
      margin-bottom: ${theme.spacings.xsmall};
      font-family: ${theme.font.highlight};
    }

    ${ButtonStyles} {
      background: ${theme.colors.primary.main};
      border: none;

      span,
      svg {
        color: ${theme.colors.complementary.light};
      }
    }

    ${ButtonStyles}:hover {
      background: ${theme.colors.primary.dark};

      span,
      svg {
        color: ${theme.colors.complementary.main};
      }
    }
  `}
`
