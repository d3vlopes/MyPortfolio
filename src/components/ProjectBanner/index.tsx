import { OpenFolder } from '@styled-icons/fluentui-system-filled/OpenFolder'

import Heading from 'components/Heading'
import Button from 'components/Button'

import * as S from './styles'

export type ProjectBannerProps = {
  src: string
  projectName: string
  projectUrl: string
}

const ProjectBanner = ({
  src,
  projectName,
  projectUrl,
}: ProjectBannerProps) => {
  return (
    <>
      <S.Cover
        src={src}
        role="image"
        aria-label="Imagem de destaque do projeto"
        data-testid="cover"
      >
        <Heading>{projectName}</Heading>
        <Button
          as="a"
          role="button"
          href={projectUrl}
          target="_blank"
          variant="outline"
          icon={<OpenFolder />}
        >
          Visitar
        </Button>
      </S.Cover>
    </>
  )
}

export default ProjectBanner
