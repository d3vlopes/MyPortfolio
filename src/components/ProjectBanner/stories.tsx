import { Story, Meta } from '@storybook/react'
import ProjectBanner, { ProjectBannerProps } from '.'

export default {
  title: 'Components/ProjectBanner',
  component: ProjectBanner,
  parameters: {
    layout: 'fullscreen',
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/KIQWZ47GDrrNCIxysbEqOA/My-Portfolio?node-id=1167%3A1033',
    },
  },
  args: {
    src: '/img/project/my-portfolio.png',
    projectName: 'Won Games',
    projectUrl: 'https://www.google.com',
  },
} as Meta<ProjectBannerProps>

export const Default: Story<ProjectBannerProps> = (args) => {
  return (
    <div>
      <ProjectBanner {...args} />
    </div>
  )
}
