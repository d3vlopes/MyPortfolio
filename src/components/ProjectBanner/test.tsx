import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import ProjectBanner, { ProjectBannerProps } from '.'

const props: ProjectBannerProps = {
  src: '/img/project/banner.png',
  projectName: 'Project Name',
  projectUrl: 'https://www.google.com',
}

const { getByRole, getByTestId } = screen

describe('<ProjectBanner />', () => {
  it('should render ProjectBanner', () => {
    const { container } = renderWithTheme(<ProjectBanner {...props} />)

    const cover = getByTestId('cover')
    const projectName = getByRole('heading', { name: 'Project Name' })
    const buttonVisitProject = getByRole('button', { name: 'Visitar' })

    expect(cover).toHaveAttribute('src', '/img/project/banner.png')
    expect(projectName).toBeInTheDocument()
    expect(buttonVisitProject).toHaveAttribute('href', 'https://www.google.com')
    expect(container.firstChild).toMatchSnapshot()
  })
})
