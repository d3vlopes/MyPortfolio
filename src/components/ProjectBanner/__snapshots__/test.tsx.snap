// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectBanner /> should render ProjectBanner 1`] = `
.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: Gilroy-Bold;
  max-width: 100%;
  font-size: 3.6rem;
  line-height: calc(3.6rem + 1rem);
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  -webkit-transition: background 0.1s ease-in-out,-webkit-transform 0.3s ease-in-out;
  -webkit-transition: background 0.1s ease-in-out,transform 0.3s ease-in-out;
  transition: background 0.1s ease-in-out,transform 0.3s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  border-radius: 0.6rem;
  background-color: transparent;
  border: 1px solid #DEE7EA;
  font-size: 2.2rem;
  width: 21.5rem;
  height: 6.2rem;
  border-radius: 1.2rem;
  height: 5rem;
}

.c4:focus {
  outline: 1px dashed #DEE7EA;
}

.c4:hover {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.c4:hover svg {
  color: #6D59A8;
}

.c4 span {
  font-family: Rubik,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-weight: 500;
  line-height: 100%;
  color: #FFFFFF;
  margin-left: 0.8rem;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
}

.c4:hover {
  border: 1px solid #6D59A8;
}

.c4:hover span {
  color: #6D59A8;
}

.c4 svg {
  width: 2.4rem;
  height: 2.4rem;
}

.c4 svg {
  color: #DEE7EA;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  height: 60rem;
  background-image: linear-gradient( 177.39deg,rgba(0,0,0,0.2) 45%,rgba(0,0,0,0) 100% ),url(/img/project/banner.png);
  background-size: cover;
  background-position: top center;
  background-repeat: no-repeat;
}

.c0 .c1 {
  margin-bottom: 1.6rem;
  font-family: CodeNext-ExtraBold;
}

.c0 .c3 {
  background: #6D59A8;
  border: none;
}

.c0 .c3 span,
.c0 .c3 svg {
  color: #F7FDFF;
}

.c0 .c3:hover {
  background: #543F8C;
}

.c0 .c3:hover span,
.c0 .c3:hover svg {
  color: #DEE7EA;
}

@media (max-width:768px) {
  .c0 {
    background-size: 100% 100%;
  }
}

<div
  aria-label="Imagem de destaque do projeto"
  class="c0"
  data-testid="cover"
  role="image"
  src="/img/project/banner.png"
>
  <h1
    class="c1 c2"
  >
    Project Name
  </h1>
  <a
    class="c3 c4"
    href="https://www.google.com"
    role="button"
    target="_blank"
  >
    <svg
      aria-hidden="true"
      class="c5"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M34.63 6.13a7.5 7.5 0 0 1 7.5 7.5v9a1.5 1.5 0 0 1-3 0v-9a4.5 4.5 0 0 0-4.5-4.5H13.36a4.5 4.5 0 0 0-4.5 4.5v21a4.5 4.5 0 0 0 4.5 4.5h9.26a1.5 1.5 0 0 1 0 3h-9.25a7.5 7.5 0 0 1-7.5-7.5v-21a7.5 7.5 0 0 1 7.5-7.5h21.24z"
      />
      <path
        d="M18 19.5c0-.83.67-1.5 1.5-1.5h13a1.5 1.5 0 0 1 0 3h-9.38l14.44 14.44a1.5 1.5 0 0 1-2.12 2.12L21 23.12v9.38a1.5 1.5 0 0 1-3 0v-13z"
      />
    </svg>
    <span>
      Visitar
    </span>
  </a>
</div>
`
